android/jni/app-android.cpp

```diff
@@ -29,7 +36,351 @@
 
 #include "app-android.h"
 
-static JNIEnv *jniEnvUI;
+// HACK: Use NativeQueryConfig
+#include "Core/Config.h"
+
+
+// Remember that all of these need initialization on init! The process
+// may be reused when restarting the game. Globals are DANGEROUS.
+
+float dp_xscale = 1.0f;
+float dp_yscale = 1.0f;
+
+InputState input_state;
+
+static bool first_lost = true;
+static std::string library_path;
+
+struct saved_state {
+	int x;
```

Comment:
Eh, this is just stuff from the sample I didn't clean out properly. I don't think we have much use for the Android activity state save thing.